invenio_vocabularies/cli.py

```diff
@@ -131,7 +132,7 @@ def update(vocabulary, filepath=None, origin=None):
     config = get_config_for_ds(vocabulary, filepath, origin)
 
     for w_conf in config["writers"]:
-        w_conf["args"]["update"] = True
+        w_conf.setdefault("args", {})["update"] = True
```

Comment:
Could this lead to unwanted behavior? 

if the `w_conf["args"]` fails the cli shows the user directly, that the configuration is wrong, but if `setdefault`adds the `args` magically the user didn't know that the `identity`should be set and at best it fails somewhere else or it silently does something the user doesn't want.

Reply:
I agree that using `setdefault` may hide the cause of missing args. What do you think is a better method to handle this you think?  
I can see the [writer_config](https://github.com/inveniosoftware/invenio-vocabularies/blob/a180d29f38f366b37d228314accac96fc74b6d81/invenio_vocabularies/cli.py#L64) coming from [DATASTREAM_CONFIG](https://github.com/inveniosoftware/invenio-vocabularies/blob/a180d29f38f366b37d228314accac96fc74b6d81/invenio_vocabularies/contrib/names/datastreams.py#L118-L125) include [args](https://github.com/inveniosoftware/invenio-vocabularies/blob/a180d29f38f366b37d228314accac96fc74b6d81/invenio_vocabularies/contrib/names/datastreams.py#L121C13-L121C19), should we add `"update": False` to all [DATASTREAM_CONFIG](https://github.com/search?q=repo%3Ainveniosoftware%2Finvenio-vocabularies%20DATASTREAM_CONFIG%20&type=code) instead?
This change seems to restore functionality, allowing successful vocabulary updates as well. 
But regarding the question of why args is been stipped out while it's there in the config it's not clear for me.

Reply:
I would rather try to be explicit in the documentation about the arguments rather than setdefault or assume default arguments. I agree that it might lead to unexpected results, and in the cli commands (which are for developer user) we would probably like to avoid unexpected behaviour.

Reply:
Got it! the documentation has been updated accordingly @kpsherva
Then we can close this one once [609](https://github.com/inveniosoftware/docs-invenio-rdm/pull/609) is been merged.